thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1368,6 +1369,15 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
             keyManagers.reset(new std::vector<std::atomic<IKeyManager *>>(parts.size()));
             for (auto & k: *keyManagers)
                 k = nullptr;
+            contextLoggers.clear();
+            for (unsigned i=0; i<parts.size(); i++)
+                contextLoggers.push_back(new CStatsContextLogger(jhtreeCacheStatistics, thorJob));
+        }
+        virtual void reset() override
+        {
+            PARENT::reset();
+            for (auto contextLogger: contextLoggers)
+                contextLogger->reset();
```

Comment:
is this correct? has it been tested? 
i.e. what happens when this is called (because KJ is in a child query or loop), are the total stats correct?